### PR TITLE
Add new route to view parked messages in PersistentSubscriptionController

### PR DESCRIPTION
**Pull request by Lougarou**
Added: New route in PersistentSubscriptionController to view parked messages /subscriptions/viewparkedmessages/{stream}/{group}

Fixes https://github.com/EventStore/EventStore/issues/2305
To be used from UI https://github.com/EventStore/EventStore.UI/pull/242

### CODE REVIEW

File: src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs [hunk 1]
```diff
@@ -63,6 +65,116 @@ public class PersistentSubscriptionController : CommunicationController {
...
+				return true;
+		
+			if (string.Equals(onlyLeader, "True", StringComparison.OrdinalIgnoreCase) ||
+			    string.Equals(onlyMaster, "True", StringComparison.OrdinalIgnoreCase)) {
+				requireLeader = true;
+				return true;
+			}
+
+			return string.Equals(onlyLeader, "False", StringComparison.OrdinalIgnoreCase) ||
+			       string.Equals(onlyLeader, "False", StringComparison.OrdinalIgnoreCase);
```
Comment (pgermishuys) [inline src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs:144]: The following should likely be...

```suggestion
			       string.Equals(onlyMaster, "False", StringComparison.OrdinalIgnoreCase);
```

We can check the AtomController as well and either fix it in this PR or raise an issue to ensure that we don't have more of this logic lying around.

File: src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs [hunk 2]
```diff
@@ -63,6 +65,116 @@ public class PersistentSubscriptionController : CommunicationController {
...
+			Register(service, "/subscriptions/viewparkedmessages/{stream}/{group}?embed={embed}", HttpMethod.Get,
+				ViewParkedMessages, Codec.NoCodecs, AtomCodecs, WithParameters(Operations.Subscriptions.Statistics));
+		}
+		private void ViewParkedMessages(HttpEntityManager http, UriTemplateMatch match) {
+			var stream = match.BoundVariables["stream"];
+			var groupname = match.BoundVariables["group"];
+			var evNum = match.BoundVariables["event"];
+			var cnt = match.BoundVariables["count"];
+
+			stream = "$persistentsubscription-" + stream + "::" + groupname+"-parked";
```
Comment (pgermishuys) [inline src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs:80]: We should have other places that build up this key and we should be able to centralize this logic. Just have a look if it's feasible to do this cleanup in this PR.

Comment (Lougarou) [reply src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs]: Thanks, added a BuildSubscriptionParkedStreamName which uses previous build function BuildSubscriptionGroupKey

File: src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs (hunk 1 above)
Comment (Lougarou) [reply src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs]: Thanks! Looks like they were only in Admin and Atom controllers. Made the fixes.

File: src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs [hunk 3]
```diff
@@ -63,6 +66,175 @@ public class PersistentSubscriptionController : CommunicationController {
 				WithParameters(Operations.Subscriptions.ProcessMessages), AckMessages);
 			RegisterUrlBased(service, "/subscriptions/{stream}/{subscription}/nack?ids={messageids}&action={action}",
 				HttpMethod.Post, WithParameters(Operations.Subscriptions.ProcessMessages), NackMessages);
+			//map view parked messages
+			Register(service, "/subscriptions/viewparkedmessages/{stream}/{group}/{event}/backward/{count}?embed={embed}", HttpMethod.Get,
+				ViewParkedMessagesBackward, Codec.NoCodecs, AtomCodecs, WithParameters(Operations.Subscriptions.Statistics));
+			Register(service, "/subscriptions/viewparkedmessages/{stream}/{group}?embed={embed}", HttpMethod.Get,
+				ViewParkedMessagesBackward, Codec.NoCodecs, AtomCodecs, WithParameters(Operations.Subscriptions.Statistics));
```
Comment (pgermishuys) [inline src/EventStore.Core/Services/Transport/Http/Controllers/PersistentSubscriptionController.cs:73]: We should be able to safely remove this endpoint now
```suggestion
```

**Comment by pgermishuys** [comment]
@Lougarou would you mind resolving the merge conflicts and then this should be good to get merged.

**Comment by hayley-jean** [comment]
It also might be a good idea to update the UI submodule as part of this PR, as the UI changes have been merged already.

**Pull request merged**